src/poker/EquityEvaluator.java

```diff
@@ -41,138 +41,101 @@ public void calculateEquity() {
-                }
-        }
-        System.out.println(calculateEquityFunckja(player1EquityTurn,player2EquityTurn,ties).get(0)); //todo slabo ze 2 razy wolasz ta funkcje bo 2 razy musi sie to liczcyc wloz ja do zmiennej a pozniej rob na niej .get(0)/.get(1)
-        System.out.println(calculateEquityFunckja(player1EquityTurn,player2EquityTurn,ties).get(1));
+        for (int i = 0; i < deckWithoutHandAndBoard.size(); i++) {
+            List<Card> possibleBoard = new ArrayList<>(flopAndTurn);
+            possibleBoard.add(deckWithoutHandAndBoard.get(i));
+            ResultHandOut resultPlayer1 = winningHand(hand, possibleBoard);
+            ResultHandOut resultPlayer2 = winningHand(hand2, possibleBoard);
+            GameResult result = getPlayersWins(resultPlayer1, resultPlayer2);
+            player1EquityTurn = player1EquityTurn + result.getPlayer1Wins();
+            player2EquityTurn = player2EquityTurn + result.getPlayer2Wins();
+            ties = ties + result.getTies();
+        }
+        List<Double> resultPlayers = finalCalculateEquity(player1EquityTurn, player2EquityTurn, ties);
+        System.out.println(resultPlayers.get(0));
+        System.out.println(resultPlayers.get(1));
     }
 
-    public void calculateEquityFlop(List<Card> flop){
+    public void calculateEquityFlop(List<Card> flop) {
         List<Card> deckWithoutHandAndBoard = new ArrayList<>();
 
         double player1EquityFlop = 0;
         double player2EquityFlop = 0;
         int ties = 0;
-       deckWithoutHandAndBoard = prepareDeckWithOutBoard(flop);
-
-        for (int i = 0; i < deckWithoutHandAndBoard.size(); i++){
-            for (int j = i + 1; j < deckWithoutHandAndBoard.size(); j++){
-                List<Card> possibleBoard= new ArrayList<>();//todo uzyj new ArrayList<>(flop) albo possibleBoard.addAll(flopAndTurn)
-                possibleBoard.add(flop.get(0));
-                possibleBoard.add(flop.get(1));
-                possibleBoard.add(flop.get(2));
+        deckWithoutHandAndBoard = prepareDeckWithOutBoard(flop);
+
+        for (int i = 0; i < deckWithoutHandAndBoard.size(); i++) {
+            for (int j = i + 1; j < deckWithoutHandAndBoard.size(); j++) {
+                List<Card> possibleBoard = new ArrayList<>(flop);
                 possibleBoard.add(deckWithoutHandAndBoard.get(i));
                 possibleBoard.add(deckWithoutHandAndBoard.get(j));
                 ResultHandOut resultPlayer1 = winningHand(hand, possibleBoard);
                 ResultHandOut resultPlayer2 = winningHand(hand2, possibleBoard);
+                GameResult result = getPlayersWins(resultPlayer1, resultPlayer2);
+                player1EquityFlop = player1EquityFlop + result.getPlayer1Wins();
+                player2EquityFlop = player2EquityFlop + result.getPlayer2Wins();
+                ties = ties + result.getTies();
+            }
+        }
+        List<Double> resultPlayers = finalCalculateEquity(player1EquityFlop, player2EquityFlop, ties);
```

Comment:
niech ta funkcja finalCalculateEquity() przyjmuje GameResult result  bedzie mniej zmiennych przez to